op-guide, media: update monitoring docs (#965)
pingcap/docs: op-guide/monitor.md

### TiDB server

- The TiDB API address: `http://${host}:${port}`
- The default port: `10080`
- The details about API names: see [TiDB HTTP API](https://github.com/pingcap/tidb/blob/master/docs/tidb_http_api.md)

The following example uses `http://${host}:${port}/status` to get the current state of the TiDB server and to determine whether the server is alive. The result is returned in the JSON format.

```bash
curl http://127.0.0.1:10080/status
{
    connections: 0,  # The current number of clients connected to the TiDB server.
    version: "5.7.25-TiDB-v3.0.0-beta-250-g778c3f4a5",  # The TiDB version number.
    git_hash: "778c3f4a5a716880bcd1d71b257c8165685f0d70"  # The Git Hash of the current TiDB code.
}
```

### PD server

- The PD API address: `http://${host}:${port}/pd/api/v1/${api_name}`
- The default port: `2379`
- The details about API names: see [PD API doc](https://cdn.rawgit.com/pingcap/docs/master/op-guide/pd-api-v1.html)

> You can use this interface to get the state of all the TiKV servers and the information about load balancing. It is the most important and frequently-used interface to get the state information of all the TiKV nodes.

Author: @dcalvin I also think this sentence is a little bit weird here. Let's remove it.

    # The details about the single TiKV node.
    {
      "store": {
        "id": 1,
        "address": "127.0.0.1:20160",
        "version": "3.0.0-beta",
        "state_name": "Up"
      },
      "status": {
        "capacity": "20 GiB",  # The total capacity.
        "available": "16 GiB",  # The available capacity.
        "leader_count": 17,
        "leader_weight": 1,
        "leader_score": 17,
        "leader_size": 17,
        "region_count": 17,
        "region_weight": 1,
        "region_score": 17,
        "region_size": 17,
        "start_ts": "2019-03-21T14:09:32+08:00",  # The starting timestamp.
        "last_heartbeat_ts": "2019-03-21T14:14:22.961171958+08:00",  # The timestamp of the last heartbeat.
        "uptime": "4m50.961171958s"
      }
    }
  ]
```

## Use the metrics interface

You can use the metrics interface to monitor the state and performance of the entire TiDB cluster.

- If you use Ansible to deploy the TiDB cluster, the monitoring system (Prometheus and Grafana) is deployed at the same time.
- If you use other deployment ways, [deploy Prometheus and Grafana](#deploy-prometheus-and-grafana) before using this interface.

> In TiDB 3.0, the Prometheus Pushgateway is **not used** by default, but you can also use it if you like. In earlier TiDB versions, the Prometheus Pushgateway is **used** by default, but you can also choose to not use it.

Reviewer: Add it as a note here with a brief introduction of ** Pushgateway**

| :-- | :-- | :-------------- |
| Node1 | 192.168.199.113| PD1, TiDB, node_export, Prometheus, Grafana |
| Node2 | 192.168.199.114| PD2, node_export |
| Node3 | 192.168.199.115| PD3, node_export |
| Node4 | 192.168.199.116| TiKV1, node_export |
| Node5 | 192.168.199.117| TiKV2, node_export |
| Node6 | 192.168.199.118| TiKV3, node_export |

1. Download the binary package.

    ```bash
    # Downloads the package.
    $ wget https://github.com/prometheus/prometheus/releases/download/v2.2.1/prometheus-2.2.1.linux-amd64.tar.gz
    $ wget https://github.com/prometheus/node_exporter/releases/download/v0.15.2/node_exporter-0.15.2.linux-amd64.tar.gz
    $ wget https://s3-us-west-2.amazonaws.com/grafana-releases/release/grafana-4.6.3.linux-x64.tar.gz

    # Extracts the package.
    $ tar -xzf prometheus-2.2.1.linux-amd64.tar.gz
    $ tar -xzf node_exporter-0.15.2.linux-amd64.tar.gz
    $ tar -xzf grafana-4.6.3.linux-x64.tar.gz
    ```

2. Start `node_exporter` on Node1, Node2, Node3, and Node4.

    ```bash
    $ cd node_exporter-0.15.2.linux-amd64

    # Starts the node_exporter service.
    $ ./node_exporter --web.listen-address=":9100" \
        --log.level="info" &
    ```

> 3. Start Prometheus in Node1.

Reviewer: ```suggestion
3. Start Prometheus for Node1.
```

        - job_name: "tidb"
            honor_labels: true # Do not overwrite job & instance labels.
            static_configs:
            - targets:
            - '192.168.199.113:10080'

        - job_name: "pd"
            honor_labels: true # Do not overwrite job & instance labels.
            static_configs:
            - targets:
            - '192.168.199.113:2379'
            - '192.168.199.114:2379'
            - '192.168.199.115:2379'

        - job_name: "tikv"
            honor_labels: true # Do not overwrite job & instance labels.
            static_configs:
            - targets:
            - '192.168.199.116:20180'
            - '192.168.199.117:20180'
            - '192.168.199.118:20180'
        ...
        ```

    2. Start the Prometheus service.

        ```bash
        # Starts Prometheus.
        $ ./prometheus \
            --config.file="./prometheus.yml" \
            --web.listen-address=":9090" \
            --web.external-url="http://192.168.199.113:9090/" \
            --web.enable-admin-api \
            --log.level="info" \
            --storage.tsdb.path="./data.metrics" \
            --storage.tsdb.retention="15d" &
        ```

> 4. Start Grafana in Node1.

Reviewer: ```suggestion
4. Start Grafana on Node1.
```

        http_port = 3000
        domain = 192.168.199.113
        [database]
        [session]
        [analytics]
        check_for_updates = true
        [security]
        admin_user = admin
        admin_password = admin
        [snapshots]
        [users]
        [auth.anonymous]
        [auth.basic]
        [auth.ldap]
        [smtp]
        [emails]
        [log]
        mode = file
        [log.console]
        [log.file]
        level = info
        format = text
        [log.syslog]
        [event_publisher]
        [dashboards.json]
        enabled = false
        path = ./data/dashboards
        [metrics]
        [grafana_net]
        url = https://grafana.net
        ...
        ```

    2. Start the Grafana service.

        ```bash
        $ ./bin/grafana-server \
            --config="./conf/grafana.ini"
        ```

### Configure Grafana

This section describes how to configure Grafana.

#### Step 1: Add a Prometheus data source

1. Log in to the Grafana Web interface.

    - Default address: [http://localhost:3000](http://localhost:3000)
    - Default account: admin
    - Default password: admin

2. Click the Grafana logo to open the sidebar menu.

> 3. Click **Data Source** in the sidebar.

Reviewer: ```suggestion
3. In the sidebar menu, click **Data Source**.
```

4. Click **Add data source**.

5. Specify the data source information.

    - Specify a **Name** for the data source.
    - For **Type**, select **Prometheus**.
    - For **URL**, specify the Prometheus address.
    - Specify other fields as needed.

6. Click **Add** to save the new data source.

#### Step 2: Import a Grafana dashboard

To import a Grafana dashboard for the PD server, the TiKV server, and the TiDB server, take the following steps respectively:

1. Click the Grafana logo to open the sidebar menu.

> 2. On the sidebar menu, click **Dashboards** -> **Import** to open the **Import Dashboard** window.

Reviewer: ```suggestion
2. In the sidebar menu, click **Dashboards** -> **Import** to open the **Import Dashboard** window.
```

3. Click **Upload .json File** to upload a JSON file (Download [TiDB Grafana configuration file](https://github.com/pingcap/tidb-ansible/tree/master/scripts)).

    > **Note:** For the TiKV, PD, and TiDB dashboards, use the `tikv_pull.json` file, the `pd.json` file, the `tidb.json` file respectively.

4. Click **Load**.

5. Select a Prometheus data source.

6. Click **Import**. A Prometheus dashboard is imported.

### View component metrics

> Click **New dashboard** in the top menu and choose the dashboard you want to view.

Reviewer: ```suggestion
In the top menu, click **New dashboard** and choose the dashboard you want to view.
```

    job = "tikv"
    ```

#### Configure PushServer

> Generally, it does not need to be configured. You can use the default port: 9091.

Reviewer 2: ```suggestion
Generally, the PushServer does not need to be configured, since the default port of `9091` can be used.
```
